### PR TITLE
Miscellaneous improvements

### PR DESCRIPTION
**Pull request by roecla**
[x] add tests that matching probabilities are square and rows add up to one. 
[x] check for duplicate index entries in params.
[x] improve dtyping

### CODE REVIEW

File: sid/contacts.py [hunk 1]
```diff
@@ -38,10 +38,11 @@ def calculate_contacts(contact_models, contact_policies, states, params, date):
             if policy_start <= date <= policy_end and cp["is_active"](states):
                 cont *= cp["multiplier"]
         if not model["is_recurrent"]:
-            cont = _sum_preserving_round(cont.to_numpy()).astype(DTYPE_N_CONTACTS)
+            cont = _sum_preserving_round(cont.to_numpy().astype(DTYPE_N_CONTACTS))
+            cont = cont.astype(DTYPE_N_CONTACTS)
         columns.append(cont)
 
-    contacts = np.column_stack(columns)
+    contacts = np.column_stack(columns).astype(float)
```
Comment (janosg) [inline sid/contacts.py:45]: Why first convert everything to `DTYPE_N_CONTACTS` and then to `float`?

File: sid/matching_probabilities.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def _get_transition_matrix_from_params(params, states, variable, model_name):
             own_prob.index = own_prob.index.get_level_values("name")
             trans_mat = _create_transition_matrix_from_own_prob(own_prob)
         else:
-            trans_mat = params.loc[loc, "value"].unstack()
+            trans_mat = params.loc[loc]["value"].unstack()
```
Comment (janosg) [inline sid/matching_probabilities.py:71]: I don't understand why this was necessary. Maybe make a comment?

File: sid/contacts.py (hunk 1 above)
Comment (roecla) [reply sid/contacts.py:45]: That doesn't make sense, you are right. But `column_stack' does not preserve the dtype of the elements passed to it. Now I just set the dtype to `DTYPE_N_CONTACTS` after `np.column_stack'.

File: sid/matching_probabilities.py (hunk 2 above)
Comment (roecla) [reply sid/matching_probabilities.py:72]: It's not necessary but saves a lot of the pandas lexsort PerformanceWarnings. I added a comment.

**Pull request merged**